### PR TITLE
Fix average energy for Wishart

### PR DESCRIPTION
**Pull request by albertpod**
This PR fixes issue #148.

### CODE REVIEW

File: src/nodes/wishart.jl [hunk 1]
```diff
@@ -13,8 +13,8 @@ import Distributions: Wishart
     m_q_out = mean(q_out)
 
     return 0.5 * (
-        m_q_ν * (logdet(m_q_S) + d * log(2)) -
-        logdet(m_q_out) * (m_q_ν - d - 1.0) +
-        tr(cholinv(m_q_S) * m_q_out) + 0.5 * d * (d - 1) * logπ
+        m_q_ν * (mean(logdet, q_S) + d * log(2)) -
+        mean(logdet, q_out) * (m_q_ν - d - 1.0) +
+        tr(mean(cholinv, q_S) * m_q_out) + 0.5 * d * (d - 1) * logπ
     ) + mapreduce(i -> loggamma(0.5 * (m_q_ν + 1.0 - i)), +, 1:d)
```
Comment (bartvanerp) [inline src/nodes/wishart.jl:19]: Are we allowed to do this? I see that it is the case in ForneyLab. So basically we compute `E_q(v)[lnGamma(v/2 - (j-1)/2)]` if I am not mistaken. Can we state `E_q(v)[lnGamma(v/2 - (j-1)/2)] = lnGamma(E_q(v)[v]/2 - (j-1)/2)`?

File: src/distributions/wishart.jl [hunk 2]
```diff
@@ -11,6 +11,12 @@ function Distributions.mean(::typeof(logdet), distribution::Wishart)
     return mapreduce(i -> digamma((ν + 1 - i) / 2), +, 1:d) + d * log(2) + logdet(S)
 end
 
+function Distributions.mean(::typeof(cholinv), distribution::Wishart)
```
[the remark this answers is not in the document]
Comment (bvdmitri) [reply src/distributions/wishart.jl:14]: Yes, @bartvanerp is right. The method definition should be `function Distribution.mean(::typeof(inv), ::Wishart)` in the same way as for [`PointMass`](https://github.com/biaslab/ReactiveMP.jl/blob/754e15c1b8b509ba52f337e5836a3aff5556a1a0/src/distributions/pointmass.jl#L92).

File: src/nodes/wishart.jl [hunk 3]
```diff
@@ -13,8 +13,8 @@ import Distributions: Wishart
     m_q_out = mean(q_out)
 
     return 0.5 * (
-        m_q_ν * (logdet(m_q_S) + d * log(2)) -
-        logdet(m_q_out) * (m_q_ν - d - 1.0) +
-        tr(cholinv(m_q_S) * m_q_out) + 0.5 * d * (d - 1) * logπ
+        m_q_ν * (mean(logdet, q_S) + d * log(2)) -
+        mean(logdet, q_out) * (m_q_ν - d - 1.0) +
+        tr(mean(cholinv, q_S) * m_q_out) + 0.5 * d * (d - 1) * logπ
```
[the remark this answers is not in the document]
Comment (bvdmitri) [reply src/nodes/wishart.jl:18]: @bartvanerp, @albertpod  `mean(inv, dist)`

File: src/distributions/wishart.jl [hunk 4]
```diff
@@ -11,6 +11,12 @@ function Distributions.mean(::typeof(logdet), distribution::Wishart)
...
+    d    = ndims(distribution)
+    ν, S = params(distribution)
+    sum([digamma.(0.5 * (ν + 1 - i)) for i in 1:d]) + d * log(2) + logdet(S)
```
Comment (bvdmitri) [inline src/distributions/wishart.jl:17]: Replace `sum([ ... ])` with `sum(...)`. Note that I just removed squared brackets:
```julia
julia> foo(n) = sum([ i for i in 1:n ])
foo (generic function with 1 method)

julia> bar(n) = sum(i for i in 1:n)
bar (generic function with 1 method)

julia> @btime foo(100)
  97.671 ns (1 allocation: 896 bytes)
5050

julia> @btime bar(100)
  1.514 ns (0 allocations: 0 bytes)
5050
```

I also not sure about `.` in `digamma.`. It is a bit confusing, I suppose this code directly ported from ForneyLab. Probably we can simply omit `.`?

Comment (bartvanerp) [reply src/distributions/wishart.jl:17]: Wow didn't know this

Comment (bvdmitri) [reply src/distributions/wishart.jl:17]: @bartvanerp The reason is this :) 
```julia
julia> typeof((i for i in 1:10))
Base.Generator{UnitRange{Int64}, typeof(identity)}

julia> typeof([i for i in 1:10])
Vector{Int64}
```
So "bracket-free" version does not allocate and generates items lazily.

**Review by bvdmitri — APPROVED** [review]

**Review by bartvanerp — APPROVED** [review]

**Pull request merged**